oidc-api/src/main/java/uk/gov/di/authentication/oidc/services/LogoutService.java

```diff
@@ -155,4 +167,30 @@ public APIGatewayProxyResponseEvent generateLogoutResponse(
         return generateApiGatewayProxyResponse(
                 302, "", Map.of(ResponseHeaders.LOCATION, uri.toString()), null);
     }
+
+    private APIGatewayProxyResponseEvent generateAccountInterventionLogoutResponse(
+            APIGatewayProxyRequestEvent input,
+            Optional<String> clientId,
+            Optional<String> sessionId,
+            AccountInterventionStatus accountStatus) {
+        String redirectUri;
+        if (accountStatus.blocked()) {
+            redirectUri = configurationService.getAccountStatusBlockedURI();
+            LOG.info("Generating Account Intervention blocked logout response");
+        } else if (accountStatus.suspended()) {
+            redirectUri = configurationService.getAccountStatusSuspendedURI();
+            LOG.info("Generating Account Intervention suspended logout response");
+        } else {
+            throw new RuntimeException("Account status must be blocked or suspended");
+        }
+        sessionId.ifPresent(t -> cloudwatchMetricsService.incrementLogout(clientId));
```

Comment:
Not sure it makes sense to associate this with the RP

Reply:
Will change the metric to include information about whether the logout was due to an RP or account interventions

Reply:
Ended up adding a metric for the account status which can be "unknown", "blocked", or "suspended". Felt a bit odd just passing "true" or "false" when we have more information available.